### ISSUE TITLE
Immutable and segmented entity content is not deleted correctly when at the edge of a selection

### ISSUE THREAD

**Issue by mmmoussa** [facebook/draft-js#1063]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
When deleting after having made a selection that contains part of an immutable or segmented entity, the entity status of the entity is correctly removed but the text within it is not deleted.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. You can use this jsfiddle to get started: https://jsfiddle.net/stopachka/m6z0xn4r/.**
Visit the draft.js entity example here: http://codepen.io/Kiwka/pen/wgpOoZ. Using the cursor, make a selection that includes part of the immutable "Superman" entity and some of the text that follows it (eg. "erman. Dele"). Click backspace.

**What is the expected behavior?**
The text within the entity should be deleted in accordance with the entity's deletion rules. In the case of the immutable entity example, all text within the entity should be deleted.

**Which versions of Draft.js, and which browser / OS are affected by this issue? Did this work in previous versions of Draft.js?**
All versions of Draft.js are affected on all browser / OS combinations. This bug did not work previously.

**Comment by flarnie** [comment]
I have confirmed that the current behavior is the intended behavior for entity deletion in the main FB use cases. I think @hellendag said something similar too. Sorry - let's look for another work-around for the issues this causes for your use case. Please open a separate issue with those more specific problems.

**Comment by brettimus** [comment]
tl;dr use this to enable this behavior (as of 0.11.0)

```
if (typeof window !== 'undefined') {
  if (!window.__DRAFT_GKX) {
    window.__DRAFT_GKX = {};
  }
  window.__DRAFT_GKX['draft_segmented_entities_behavior'] = true;
}
```

***

If anyone else stumbles across this issue, I'm using `0.11.0` and ran into essentially the same problem (I wanted `alt+backspace` to delete immutable entities that maybe had spaces in them.)

Seems like the aforementioned behavior change to support this behavior was reverted (see 361f811), BUT when I dug around in the current code, it still exists... just under a silly little feature flag (see: https://github.com/facebook/draft-js/blob/d2bd0439f100afd218b4e3f8cb1b573738d33ccd/src/model/modifier/DraftModifier.js#L175)

```
    if (gkx('draft_segmented_entities_behavior')) {
      // Adjust the selection to properly delete segemented and immutable
      // entities
      adjustedRemovalRange = getCharacterRemovalRange(
        contentState.getEntityMap(),
        startBlock,
        endBlock,
        rangeToRemove,
        removalDirection,
      );
    }
```

This `gkx` business turns out to be a helper function that checks for experimental flags. So, to make this work, you gotta know the right feature flag.

⚠️ PROCEED WITH CAUTION. Enabling draft feature flags is undocumented atm

If you wanna enable the flag that gives you this behavior, try the following:

```
if (typeof window !== 'undefined') {
  if (!window.__DRAFT_GKX) {
    window.__DRAFT_GKX = {};
  }
  window.__DRAFT_GKX['draft_segmented_entities_behavior'] = true;
}
```
